src/app.js

```diff
@@ -58,6 +58,42 @@ var HackToolboxApplication = GObject.registerClass(class extends Gtk.Application
         });
         flip.connect('activate', this._onFlip.bind(this));
         this.add_action(flip);
+
+        const init = new Gio.SimpleAction({
+            name: 'init',
+            parameter_type: new GLib.VariantType('(ss)'),
+        });
+        init.connect('activate', this._initToolbox.bind(this));
+        this.add_action(init);
+    }
+
+    _initToolbox(action, parameterVariant) {
+        this.hold();
```

Comment:
It's good practice when pairing hold/release calls to do something like this:
```js
this.hold();
try {
    doStuff();
} finally {
    this.release();
}
```
This avoids accidentally unpairing the calls in case there is an exception.

(I realize the existing code isn't doing that, but it would be good to do it in new code.)

Reply:
I've added the try-finally to the new code.